netty-reactive-streams/src/main/java/com/typesafe/netty/HandlerPublisher.java

```diff
@@ -203,32 +203,40 @@ private void provideSubscriber(Subscriber<? super T> subscriber) {
 
     @Override
     public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
-        verifyRegisteredWithRightExecutor(ctx);
+        // If the channel is not yet registered, then it's not safe to invoke any methods on it, eg read() or close()
+        // So don't provide the context until it is registered.
+        if (ctx.channel().isRegistered()) {
+            provideChannelContext(ctx);
+        }
```

Comment:
for this `channel will often have already been registered to the event loop.` you could add your own handler to the ServerChannel's channelpipeline, and then remove the first handler that netty added which will automatically register the channel to the eventloop.
by this way,you have the full controll.

```java
        p.addLast(new ChannelInitializer<Channel>() {
            @Override
            public void initChannel(Channel ch) throws Exception {
                ChannelPipeline pipeline = ch.pipeline();
                ChannelHandler handler = handler();
                if (handler != null) {
                    pipeline.addLast(handler);
                }
                pipeline.addLast(new ServerBootstrapAcceptor(
                        currentChildGroup, currentChildHandler, currentChildOptions, currentChildAttrs));
            }
        });
```
the problem is the `ServerBootstrapAcceptor` .I encounter this before,and then I remove the `ServerBootstrapAcceptor` ,after that I make use of the ServerChannel directly and bypass the `ServerBootstrap` helper class.

I think that was the root cause.

If we want this to merge into netty ,I think one approach should be 
```
class ReactiveServerBootstrap extends ServerBootstrap{
  override def init(channel: Channel) = ...//rewrite the auto accept and auto register behavior.
}
` ``

You can see,by this way,we could have the same model as `akka-io`.